services/management/mocks.go

```diff
@@ -26,4 +30,6 @@ type registry interface {
 	IsConnected(pmmAgentID string) bool
 	Kick(ctx context.Context, pmmAgentID string)
 	SendSetStateRequest(ctx context.Context, pmmAgentID string)
+	// SendRequest sends request to pmm-agent with given id.
+	SendRequest(ctx context.Context, pmmAgentID string, payload agentpb.ServerRequestPayload) (agentpb.AgentResponsePayload, error)
```

Comment:
Let's replace `SendRequest` with more specific method(s). (I guess they would be `RunAction`, `CancelAction`, `GetActionResult`).

Reply:
IMHO, It's really becoming god object. Let's keep SendRequest and extract SendSetStateRequest to a separate struct.

Reply:
If you want to keep `SendRequest`, then `SendSetStateRequest` have to be removed – they are on different levels of abstraction: `SendSetStateRequest` _uses_ `SendRequest`.

I don't think exposing `agentpb.ServerRequestPayload` and `agentpb.AgentResponsePayload` is a good idea.

Reply:
I analyzed code and I agree that there should be 3 specific methods. But they should be outside of registry in agents package. And I agree that exposing agentpb.ServerRequestPayload outside of agents package isn't a good idea. sendRequest should be a package local method.

Reply:
> I don't think exposing agentpb.ServerRequestPayload and agentpb.AgentResponsePayload is a good idea.

Agree... It's strange, that application-level service knows about infrastructure data... I'm thinking about it right now.

@AlekSi Btw I've started MySQL EXPLAIN and I've made a new branch from this one. Also I have some code with the initial implementation. 

Maybe we can merge this anyway? I want to change and refactor this thing with MySQL EXPLAIN action. At this moment I have a more clear vision than I had before. 
Also, I plan to finish EXPLAIN in 2-3 days. I mean this improvement will be easy to implement with EXPLAIN implementation.